runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -118,17 +150,15 @@ local function diagnostic_lsp_to_vim(diagnostics, bufnr, client_id)
       end_col = line_byte_from_position(buf_lines, _end.line, _end.character, offset_encoding),
       severity = severity_lsp_to_vim(diagnostic.severity),
       message = diagnostic.message,
+      related_information = related_information_lsp_to_vim(
+        diagnostic.relatedInformation,
+        offset_encoding
+      ),
       source = diagnostic.source,
       code = diagnostic.code,
       _tags = tags_lsp_to_vim(diagnostic, client_id),
       user_data = {
-        lsp = {
-          -- usage of user_data.lsp.code is deprecated in favor of the top-level code field
-          code = diagnostic.code,
-          codeDescription = diagnostic.codeDescription,
-          relatedInformation = diagnostic.relatedInformation,
-          data = diagnostic.data,
-        },
+        lsp = diagnostic,
```

Comment:
Can you also update `diagnostic_vim_to_lsp` to make use of this?

Reply:
See my top level comment: if I understand the code correctly, `diagnostic_vim_to_lsp` already makes use of this, since it already merges the contents of `user_data.lsp` into the returned table. (This is also why the unit tests currently fails, I fixed this now in the latest push)